scripts/check_copyright_header/src/main.rs

```diff
@@ -0,0 +1,150 @@
+use std::{
+    env,
+    fs::File,
+    io::{BufRead, BufReader},
+};
+
+fn check_file_header(lines: impl IntoIterator<Item = String>) -> Result<(), &'static str> {
+    let mut found_zefchain_labs_header = false;
+    let mut is_end_of_header = false;
+
+    for line in lines {
+        if !is_end_of_header {
+            if line == "// Copyright (c) Zefchain Labs, Inc." {
+                found_zefchain_labs_header = true;
+                continue;
+            }
+
+            if line.starts_with("// Copyright (c)") {
+                continue;
+            }
+
+            if line == "// SPDX-License-Identifier: Apache-2.0" {
+                is_end_of_header = true;
+                continue;
+            }
+
+            return Err("Unexpected line reached");
+        } else {
+            if !found_zefchain_labs_header {
+                return Err("Incorrect copyright header, Zefchain Labs header not found");
+            }
+
+            if line.is_empty() {
+                // Found separation line
+                return Ok(());
+            } else {
+                return Err("Separation line not found");
+            }
+        }
+    }
+
+    Err("Incorrect copyright header")
+}
+
+fn main() {
+    let args: Vec<String> = env::args().collect();
+    let file_path = args.get(1).expect("Usage: FILE");
+
+    let file = File::open(file_path).expect("Failed to open file");
+    let reader = BufReader::new(file);
+
+    let lines: Vec<String> = reader
+        .lines()
+        .collect::<Result<_, _>>()
+        .expect("Failed to read lines from file");
+
+    match check_file_header(lines) {
+        Ok(()) => {
+            std::process::exit(0);
+        }
+        Err(error) => {
+            eprintln!("Error in file {}: {}", file_path, error);
+            std::process::exit(1);
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_valid_file_with_header() {
+        let lines = vec![
+            "// Copyright (c) Zefchain Labs, Inc.".to_string(),
+            "// SPDX-License-Identifier: Apache-2.0".to_string(),
+            "".to_string(),
+            "// Rest of the file...".to_string(),
+        ];
```

Comment:
Maybe somegthing like:
`let lines = ["foo", "bar"].into_iter().map(String::from);`
or with my suggestion above:
`let lines = ["foo", "bar"].into_iter().map(String::from).map(Ok);`

Reply:
I tried `let lines = ["foo", "bar"].into_iter().map(String::from).map(Ok);` and it produces a map, but we'll want `Lines<BufReader<File>>` if I follow your suggestion